Outline

rrrutledge opened the pull request.

utanapishtim commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+We'll also share a few patterns unique to inner source rollout including:
+
+* Run your inner source program as inner source projects.
+* Do (or say) nothing save it is documented.
```
Not sure what this means.

rrrutledge replied:
We write (or code) down what we are telling others to do and what we are doing ourselves.

mrsanz replied:
The principle is more about teaching a developer to fish so they can feed themselves rather than fishing for them.

rrrutledge replied:
I think we are always teaching people to fish.  This advice is even higher up.  More like:

* Create a fishing school rather than always teaching people yourself.
* Build a fishing machine so people don't even need to fish.

mrsanz replied:
oh even better articulation!! Nice!

rrrutledge replied:
Looking again at what I'm proposing an inner source program should provide (documentation and software), you could really say that the inner source program should provide a fishing school and fishing machines.  Move above that level of teaching others to fish one-on-one.  Maybe this mode of explanation and analogy could be woven into the whole talk?

rrrutledge replied:
Maybe that's also a way of speaking against the top-down approach of getting management to "make" people do inner source.  Is doing so like fishing for them?  They should be up-leveled to fish for themselves.

And not only that, the way that you up-level them to fish for themselves is by having them attend your fishing school and use your fishing machines.  We're takin' fishing from the 19th century to the 21st century.

Does that make sense?  It was helpful for me to write it, at least.

mrsanz replied:
yea I think it would be a nice way to frame the whole conversation and keep it relatable.

rrrutledge replied:
OK, great.

utanapishtim commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+* Run your inner source program as inner source projects.
+* Do (or say) nothing save it is documented.
+* Program growth/maturity progression starts with facilitating any synchronous, in-person, side-project talking collaboration then grows to facilitating asynchronous, main-project coding collaboration.
```
I think this should be couched as a hypothesis. Do we have any substantial evidence that shows this is the case? If this is based on anecdotal evidence, I think we should be clear about that and mention the anecdotes.

rrrutledge replied:
Noted.  I can't think of textual changes to make at all, but perhaps when speaking can do so.  There are some similar disclaimers in the abstract and introduction.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+#### Outline
+
+An inner source program is the set of tools, documentation, training, and so forth provided so that any team can participate in inner source effectively as either a maintainer or contributor.
```
`and so forth`  is what precisely? omit or be more specific?

mrsanz replied:
Early on in this I think you should articulate that any program should be self sustaing.  the impulse is to go whole hog and rollout big programs etc. I think the first point should be to clearly define or establish a vision or picture of what the new culture will look like then organically grow support until you hit your early majority at which phase two is rolling out the program guns.  A program should be able to support its own weight.  While not 100% complete some ideas can be found in my value stream proposal around [core body strength](https://www.notion.so/Core-Body-Strength-c23f26451b024b52a705dc6ba46b15a4) that might help support this.

There will be problems unique to your own organization.  These tend to come in two forms.  1) In companies where there is a flat organic mindset, people tend to employ the workaround strategy.  2) In companies where there is rigid top down authority based structure there tends to be more escalation based.  This means your vision should at least identify and work to attack these two problems.   In a lot of ways these are mirrored strategies, one coming from too much centralized governance and one coming from too little centralized governance.

You can explain that we have a more flat structure and are working to combat the workaround scenario in most cases, but we too have some places where interrupt and escalation based ivory towers need some attention too.   

I think this insight is super valuable and will provide a great backdrop for the talk.

rrrutledge replied:
> and so forth is what precisely? omit or be more specific?

omitted.

> my value stream proposal around core body strength

<img width="678" alt="screen shot 2018-05-03 at 5 17 42 am" src="https://user-images.githubusercontent.com/9609562/39576008-63a10932-4e91-11e8-8580-db67bd102e39.png">

> There will be problems unique to your own organization. These tend to come in two forms

Very insightful comment!  I hadn't thought if it that way.  I like it!  I wish the Learning Path presentation came first so that I could use that terminology.  Do you think that the approach we'd take to rolling at the program would look that different based on trying to combat _workaround_ vs. _escalate_?

rrrutledge replied:
I have access to the _core body strength_ page in _notion_ now.

mrsanz replied:
It might be worth taking 5 minutes to highlight the terms for consistenecy

rrrutledge replied:
I'm having trouble with this as it seems like a point that is orthogonal to the main content.  Still thinking of leaving it out (or at least wait until the presentation to see how it fits in).  If it's something you feel strongly about then let's meet to discuss?

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+To frame the talk, we'll explain the goals of an inner source program as well as the context within-which we've begun to pursue those goals.
+Even though the learning in this presentation is occurring in that context, we suspect that much of what's here applies generally.
+
+When building an inner source program you should have a vision of what services you will provide to facilitate inner sourcing _as well as_ a strategy you will employ to roll out and make effective these services in your organization.
```
This seems backwards...  or at least perhaps you should define what a service is  before you state something like this.   This makes it sound like I should go and build a bunch of InnerSource tooling as a part of my program.

rrrutledge replied:
Removed the word `service`.  Does that help?

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+The services of the inner source program should provide two things.
+First, they must teach, define, and document the type of culture that is conducive to inner source.
+Second, there must be software that reinforces that culture while simultaneously lowering the barrier to entry to participating in that culture.
+Examples of these services include:
```
Not sure that you should provide this many examples.  I think they dilute the first two points a bit.  I think the only one we're super confident in is the playbook model.  And this facilitates this point "Software/convention to facilitate individuals communicating with one another"

I'd say that Curated training on the above software and documentation is what takes things to the next level.  Not just having a program but ensuring that it thrives.

mrsanz replied:
I think the thing missing here is the regular customer outreach and marketing that goes along with it.   I think without a regular pulse you'll end up with the flash mob problem, where the point to point connections are only as good as the person network who is facilitating the communities.   I know that we haven't defined it but it is critical to the success we've had so far.  If you stopped doing we wouldn't have our communities.  So I think you need to be able to articulate this point in your talk.  Without it you'd have people mandated to participate, which would result in a disastrous innersource program.

rrrutledge replied:
> Not sure that you should provide this many examples.  I think they dilute the first two points a bit. I think the only one we're super confident

Is the recommendation to leave them out due to the dilution angle or lack of confidence angle?  If dilution, these points are meant to be specific examples of the two points (expand on them).  If lack of confidence, I think it's OK as long as it's the kind of stuff we think we want to do.  The point (in my mind) of the presentation is to present a case for working this way so that as we do it in the upcoming months others may be interested in working with us as we go about implementing these ideas.

rrrutledge replied:
> missing here is the regular customer outreach

I think that would go further below in the section of how to roll out the program?

mrsanz replied:
My main concern is that by giving such concrete examples folks will think that they should implement the same things that we have to get an innersource program up and running.  I don't think we're ready to give that level of recommendation. 

Instead I'd propose that you focus on describing how to create the culture that will cut through [strategy](https://nikedigital.slack.com/archives/C6TC8J68L/p1525311502000026).

mrsanz replied:
>  The point (in my mind) of the presentation is to present a case for working this way so that as we do it in the upcoming months others may be interested in working with us as we go about implementing these ideas.

it seems like that should be at the end then as your call to action.  Here are some of the solutions we've started pursuing....

rrrutledge replied:
Sounds good.  There is a sentence at the end about hoping to work together.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+We'll review real examples of how these patterns played out in our journey.
+We'll also share a few patterns unique to inner source rollout including:
+
+* Run your inner source program as inner source projects.
```
or "Eat your own dogfood!"

mrsanz replied:
I think this is too generic on its own.  Seems like people following this talk will run into the chicken-egg problem.  How do you run a IS program as a set of IS projects if there is not a program to define what that means.  I'd like to see more concrete and actionable things here.  

Like create a community of leaders?  Run something like our Tuesday sessions?  Or more along the lines of the engineering forum cop?

rrrutledge replied:
Is it fair to say that you feel that this point is valid to include in the outline and you're looking forward to seeing how it's expounded in the actual presentation?

In my mind this means something like our Tuesday sessions and efforts.

mrsanz replied:
I think I do think it is valid, but I'd like to know how you are going to reinforce it, because it can go too many different directions as it is currently written.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+Examples of these services include:
+
+* Documentation on how to do inner source (and relation of that documentation to [InnerSourcePatterns](http://innersourcecommons.org/patterns)).
+* Software for automated, two-way discovery of inner source projects (maintainer to contributor/user and contributor/user to maintainer).
```
Scout seems like a one way discovery mechanism and its a bit clunky to discover InnerSource projects.  I'm not sure we can lean on this bullet yet.  Are we describing the ideal state or the state where we are today?

rrrutledge replied:
ideal state.  The point (in my mind) of the presentation is to present a case for working this way so that as we do it in the upcoming months others may be interested in working with us as we go about implementing these ideas.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+* Documentation on how to do inner source (and relation of that documentation to [InnerSourcePatterns](http://innersourcecommons.org/patterns)).
+* Software for automated, two-way discovery of inner source projects (maintainer to contributor/user and contributor/user to maintainer).
+* Software for a data-driven understanding of the impact of inner source projects.
```
All I can point to is our weekly metrics. We don't do this. I think the valuation topic is important but i don't know that software is mature enough to talk about this yet.

I think that based on what you said this talk is about then we should point to things that we've been doing not what we hope to do.

rrrutledge replied:
Oh, I was thinking it's useful to talk about what we hope to do as well.  Maybe we should align on that point, here.

mrsanz replied:
I don't think that we should talk about futures.  Remember the point of the talk is to get others understanding what is essential to run an InnerSource program.  Following the principle of **_Just Enough_** means that futures can come well in the future ;)

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+* Software for automated, two-way discovery of inner source projects (maintainer to contributor/user and contributor/user to maintainer).
+* Software for a data-driven understanding of the impact of inner source projects.
+* Software/convention to facilitate individuals communicating with one another.
+* Curated training on the above software and documentation.
```
As I mentioned earlier I think that this is part of the next phase, and that this isn't needed to have a successful InnerSource program.  

But it is needed to drive additional value into your program to raise the boats which is really helpful in ensuring that it is sustainable.

rrrutledge replied:
It's an interesting point to call out what is an MVP vs. the final program.  Maybe the feedback here is to make that more clear throughout the presentation (yes)?

mrsanz replied:
yes

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+* Do (or say) nothing save it is documented.
+* Program growth/maturity progression starts with facilitating any synchronous, in-person, side-project talking collaboration then grows to facilitating asynchronous, main-project coding collaboration.
+
+After explaining these principles with clear words and real anecdotes, we'll summarize the current state of our inner source program, where we're hoping it will go, and how we aim to get there. We'll express our hope to learn from and work with everyone present via open dialogue and sharing in the inner source commons.
```
There are principles missing from this list. I think that some of these are overlook as they are part of the teaching, defining and documenting the culture part.  However I think there are principles that are overlooked because they seem obvious to us.  

- Precise discovery mechanisms and patterns for shared knowledge and shared artifacts.
- Systematic and regular outreach for "drivers".
- Creating and using an economic argument for the value of community.
- Defining easy and reusable patterns (that are documented)
- Zen like action, never moving in a direction that creates waste or more than is needed.  Following the (Just Enough) principle.

rrrutledge replied:
Is it fair to say that those are included (at an outline level) in this statement:

> Documented patterns in this area (well-summarized in [the Fearless Change series](http://www.fearlesschangepatterns.com/)) apply and work

And in the actual presentation we can include exposition of the ideas you've listed out specifically?

mrsanz replied:
Sure ok!  Yea I didn't get much out of the statement referencing the Fearless Change series.  I'm not sure what level of experience your audience will have with that book, and the while the patterns describe behaviors, I'm not sure that they all describe actionable steps to take.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+* Understand the suggested purposes and principles of an inner source program.
+* Understand the suggested component parts of an inner source program and their principles.
+* Gain ideas on how to roll out an inner source program.
```
I like this one.  I think you should put it first since it has the most immediate value to your audience.

rrrutledge replied:
Sure.

mrsanz commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+After attending this presentation, we hope that viewers will:
+
+* Understand the suggested purposes and principles of an inner source program.
+* Understand the suggested component parts of an inner source program and their principles.
```
This might be too much to go into?  I'd rather you focus on clearly articulating "suggested purposes and principles" than getting into the weeds on the things that fall out from that e.g. "component parts" 

I'm saying I think it is ok to share ideas and what we've started doing but I don't think that should be the emphasis of your talk.  So rather than having them understand this as a result of your talk, I'd rather see them have a framework for talking about potential component parts so that proposals for atomic parts can begin to surface in the community as we have begun to do with our value stream work.

This will lead to more collaboration for the last two takeaways in your list.

rrrutledge replied:
The goal is definitely more collaboration, I guess I thought that they may need to know some specifics on what we're doing to be interested in collaborating on it.  Do you feel they'd get enough motivation just by understanding the general principles we're trying to follow?

I feel like we need to seed the idea that there are some specific projects that we all will need in order to get everyone's brains going on the idea of working on it together.

utanapishtim replied:
I like the measurable goal of this presentation as actual collaboration with other teams in other companies on our work. That's definitely something we can measure! It's also a really exciting idea! Excited to see the presentation now with this goal in mind!

mrsanz commented:
[This slack message](https://nikedigital.slack.com/archives/C6TC8J68L/p1525299716000445) seems generally applicative as well.

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+## Short Description
+
+"Inner source is a great idea! Go do inner source!"
```
This example is good, and I want to find the place to use it.  Can I just keep it up my sleeve to use in the actual talk but leave it out of these descriptions?  Nowhere else here have I actually specified _how_ I am communicating any of this points - only _what_ is being communicated.

I did add this extra sentence to the detailed description to capture raw essence of the point that we want to make:

`Participation in inner source requires real amounts of care and feeding to be effective.`

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+How do you realistically affect the behavior of dozens or hundreds of teams to the point where robust and pervasive inner sourcing is a normal part of the way that engineering is done?
+
+This presentation shares principles, practical anecdotes, and relatable examples on this situation gleaned from experience at Nike.
+While not claiming all of the answers, it is insightful summary of over a year of work in the area.
```
`While not claiming all of the answers, it summarizes the learnings of over a year of work in the area.`

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+This presentation shares principles, practical anecdotes, and relatable examples on this situation gleaned from experience at Nike.
+While not claiming all of the answers, it is insightful summary of over a year of work in the area.
+
+## Audience
```
You are an insightful individual!  The short description is already posted as the talk abstract on [the conference page](http://paypal.github.io/InnerSourceCommons/events/isc-spring-2018-agenda/).

I moved the **Audience** section up.

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+## Audience
+This presentation will be given at [the ISC Spring Summit](http://paypal.github.io/InnerSourceCommons/events/isc-spring-2018-agenda/).
+It is intended for those that are evangelizing inner source in a specific company.
```
✅

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+#### Outline
+
+An inner source program is the set of tools, documentation, training, and so forth provided so that any team can participate in inner source effectively as either a maintainer or contributor.
```
Sounds good.

rrrutledge commented on outline.md:
```diff
@@ -0,0 +1,59 @@
+
+An inner source program is the set of tools, documentation, training, and so forth provided so that any team can participate in inner source effectively as either a maintainer or contributor.
+To frame the talk, we'll explain the goals of an inner source program as well as the context within-which we've begun to pursue those goals.
+Even though the learning in this presentation is occurring in that context, we suspect that much of what's here applies generally.
```
Well, I want to be clear from the beginning that we're not just presenting a case study but want to make recommendations on how an inner source program should be done.  Is that good?

rrrutledge commented:
Added this line to capture the iterative possibility of building an inner source program: `You can start the initiative with _any_ of the aforementioned services and add more services to it as needed over time`.  As I thought about it I believe that you really can start _anywhere_ and add on to the program in any order that works for you.

utanapishtim commented:
LGTM.

mrsanz commented:
Last bits of in person feedback.  

Driving force of the program should be articulated.
- "Opt in" nature seems to be a principle that should be explained a bit

rrrutledge commented:
> Driving force of the program should be articulated.

Added in the specific patterns from _Fearless Change_ which I think apply.  There were also two general principles that I think are important nevertheless do not have reference in _Fearless Change_.  I added them directly and called them **Opt In** and **Grass Roots**.
